crates/polars-stream/src/async_executor/task.rs

```diff
@@ -278,6 +278,10 @@ impl<M> Runnable<M> {
 
 pub struct JoinHandle<T>(Option<Arc<dyn Joinable<T>>>);
 pub struct CancelHandle(Weak<dyn Cancellable>);
+pub struct AbortOnDropHandle<T> {
```

Comment:
This smells to me, can you explain why you would need this?

Reply:
I'm spawning tasks in some places with buffering, and the task handles get put into buffers/channels. For the case where the source node is shut down early (e.g. due to slice limit), those buffers may still contain task handles when dropped. If those handles aren't wrapped with `AbortOnDrop` they would still continue to be scheduled and run to completion, even though we don't need those results anymore.